compiler/il/OMRSymbol.hpp

```diff
@@ -398,8 +393,32 @@ class OMR_EXTENSIBLE Symbol
    inline void setMemoryTypeShadowSymbol();
    inline bool isMemoryTypeShadowSymbol();
 
-   void setOrdered() { _flags.set(Ordered); }
-   bool isOrdered()  { return _flags.testAny(Ordered); }
+   enum MemoryOrdering
+      {
+      TransparentSemantics,
+      OptOpaqueSemantics,
+      AcqRelSemantics,
```

Comment:
I would suggest spelling this out as `AcquireReleaseSemantics`, along with the names of the `set` and `is` methods.